### PR TITLE
cluster, schedule: unify the scheduling halt to decouple dependencies

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #5839. As the comment https://github.com/tikv/pd/pull/6498#discussion_r1218852310 said, unify the scheduling halt to decouple dependencies between `cluster` and `coordinator`.

### What is changed and how does it work?

```commit-message
Unify the scheduling halt to decouple dependencies between `cluster` and `coordinator`.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/coordinator.go [hunk 1]
```diff
@@ -145,7 +145,7 @@ func (c *Coordinator) PatrolRegions() {
 			log.Info("patrol regions has been stopped")
 			return
 		}
-		if allowed, _ := c.cluster.CheckSchedulingAllowance(); !allowed {
+		if !c.isSchedulingHalted() {
```
Comment (rleungx) [inline pkg/schedule/coordinator.go:148]: Maybe `c.isSchedulingHalted()`?

File: server/config/persist_options.go [hunk 2]
```diff
@@ -920,14 +920,26 @@ func (o *PersistOptions) SetAllStoresLimitTTL(ctx context.Context, client *clien
 	return err
 }
 
+var haltSchedulingStatus = schedulingAllowanceStatusGauge.WithLabelValues("halt-scheduling")
+
 // SetHaltScheduling set HaltScheduling.
-func (o *PersistOptions) SetHaltScheduling(halt bool) {
+func (o *PersistOptions) SetHaltScheduling(halt bool, source string) {
```
[the remark this answers is not in the document]
Comment (JmPotato) [reply server/config/persist_options.go:926]: Though it's only used for the metrics and may be lost due to the crash, it's OK for me not to persist it in the PD. WDYT?

[the remark this answers is not in the document]
Comment (rleungx) [reply server/config/persist_options.go:926]: It's ok to me.

**Review by rleungx — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**